### DISCUSSION TITLE
Create Example object for Classification with SPacy 3.x

### DISCUSSION THREAD

**Discussion by nicolalandro** [explosion/spaCy#9850] [Help: Coding & Implementations] [answered]
How to create correctly the Example? I try with this code:
```
 doc = nlp.make_doc(text)
examples = [Example.from_dict(doc, { "tags": [annotation] } )]
```
But it get the following error:
```
ValueError: [E971] Found incompatible lengths in `Doc.from_array`: 1 for the array and 7 for the Doc itself.
```
How I should create it?

**Comment by DuyguA** [comment, marked as answer]
> How to create correctly the Example? I try with this code:
> 
> ```
>  doc = nlp.make_doc(text)
> examples = [Example.from_dict(doc, { "tags": [annotation] } )]
> ```
> 
> But it get the following error:
> 
> ```
> ValueError: [E971] Found incompatible lengths in `Doc.from_array`: 1 for the array and 7 for the Doc itself.
> ```
> 
> How I should create it?

Thanks for your question!

There should be one tag per token, hence Doc length and annotation length should match. Here's an example, there are 2 tokens in my Doc:

```
>>> text = "you go"
>>> doc = nlp.make_doc(text)
>>> annotation = ["PRON", "VERB"]
>>> example = Example.from_dict(doc, {"tags": annotation})
```

**Reply by DuyguA** [reply]
Keep in mind that tokens are assigned by the Tokenizer (so don't count whitespaces while calculating the Doc length :slightly_smiling_face: ). For an example contracted forms are tokens on their own:

```
>>> doc = nlp.make_doc("It's a new day")
>>> [token for token in doc]
[It, 's, a, new, day]
>>> len(doc)
5

```

**Reply by nicolalandro** [reply]
But I want to do Text Classification and not Named entity recognition, so I have One class for whole string. do I make something different?

**Reply by DuyguA** [reply]
Ah OK, I saw a `tags` in your question and I thought you wanna make POS training. Then you can do the following 

for `binary textcat`:

```
label = {"TOXIC": True}
example = Example.from_dict(nlp.make_doc(text), {"cats": label})
```
Above you want to make a binary classification and your label is called `TOXIC` .

for `multi_textcat`:

```
label = {"FIVE_STAR": True, "FOUR_STAR": False, "THREE_STAR": False}
example = Example.from_dict(nlp.make_doc(text), {"cats": label})
```

Here is similar, you want to make multilabel classification and your labels are `FIVE_STAR` , `FOUR_STAR` and `THREE_STAR`.

**Reply by nicolalandro** [reply]
Ok thank you perfect! But now I have another problem during update:
```
KeyError: "Parameter 'E' for model 'hashembed' has not been allocated yet."
```
I do the following:
```
import spacy
import spacy.cli
from spacy.pipeline.textcat import single_label_cnn_config, Config
from spacy.training.example import Example

nlp=spacy.load('it_core_news_lg')

from spacy.util import minibatch, compounding

# Disabling other components
other_pipes = [pipe for pipe in nlp.pipe_names if pipe != 'textcat']
with nlp.disable_pipes(*other_pipes): # only train textcat
    optimizer = nlp.resume_training()

# Performing training
for epoch in range(5):
    losses = {}
    batches = minibatch(train, size=compounding(4., 32., 1.001))
    for batch in batches:
        examples = [Example.from_dict(nlp.make_doc(text), {"cats": {annotation:True}}) for text, annotation in batch]
        nlp.update(examples, losses=losses, drop=0.3)
```

**Comment by DuyguA** [comment]
I think the issue is excluding the tok2vec pipeline component, `hashembed` is part of that component. Can we change your code a bit : 

```
include_pipes = ["tok2vec", "textcat"]
other_pipes = [pipe for pipe in nlp.pipe_names if pipe not in include_pipes]
```
